nservicebus/upgrades/throughput-tool.md

```diff
@@ -0,0 +1,143 @@
+
+
+### Azure Service Bus
+
+Collecting metrics from Azure Service Bus relies upon the Azure Command Line Interface (CLI), which must be installed first.
+
+1. Install a version of [Powershell or Powershell Core](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell) on the host system.
+1. Install the [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
+1. From a command line, execute `az login`, which will open a browser to complete the authentication to Azure. The Azure login must have access to view metrics data for the Azure Service Bus namespace.
+1. Execute `az set --subscription {SubscriptionId}`, where `SubscriptionId` is a Guid matching the subscription id that contains the Azure Service Bus namespace.
+1. In the Azure Portal, go to the Azure Service Bus namespace, click **Properties** in the side navigtation, and then copy the `Id` value, which will be needed to run the tool. The `Id` value should have a format similar to `/subscriptions/{Guid}/resourceGroups/{rsrcGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}`.
+
+Once these prerequisites are comlpete, execute the tool with the resource id of the Azure Service Bus namespace, as in this example:
+
+```shell
+particular-throughput-tool azureservicebus --resourceId /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/my-resource-group/providers/Microsoft.ServiceBus/namespaces/my-asb-namespace --outputPath throughput-report.json
+```
+
+The tool may open additional terminal windows, which are Powershell processes gathering the data from the Azure CLI.
+
+Unlike other query options, using Azure Service Bus metrics allows the tool to capture the last 30 days worth of data, which means that results are returned instantly.
+
+#### Options
+
+All options are required:
+
+| Option | Description |
+|-|-|
+| <nobr>`--resourceId`</nobr> | The resource ID of the Azure Service Bus namespace, which can be found in the Azure Portal as described above. |
+| <nobr>`--outputPath`</nobr> | A local path where the JSON report file should be generated. |
+
+### RabbitMQ
+
+Once installed, execute the tool with the RabbitMQ management URL, as in this example:
+
+```shell
+particular-throughput-tool rabbitmq --apiUrl http://localhost:15672 --outputPath throughput-report.json
+```
+
+The tool will prompt for the username and password to access the RabbitMQ management interface. After that, it will take its initial reading, then sleep for 24 hours before taking its final reading and generating a report.
+
+#### Options
+
+All options are required:
+
+| Option | Description |
+|-|-|
+| <nobr>`--apiUrl`</nobr> | The URL for the RabbitMQ management site. |
+| <nobr>`--outputPath`</nobr> | A local path where the JSON report file should be generated. |
+
+### SQL Transport
+
+Once installed, execute the tool with the database connection string used by SQL Server endpoints, as in this example:
```

Comment:
Could we point them to some docs as to where they could find the connection string?

Also, is there any validation that is done to see that we're looking at the "right" database, like checking for the existence of queue tables.